### ISSUE TITLE
Publish Autosubmit 4.1.9 to DockerHub

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1376]
Related to #1366, which has already been partially addressed with AS 4.1.9 published to PyPI. This issue is to update the Docker recipe, and to upload it to DockerHub.

@LuiggiTenorioK did some useful improvements to our `Dockerfile` in !426.

@edgano, for the previous image, 4.1.7, I used Luiggi's code, applying a patch manually and then publishing the image (we were in a hurry due to a deliverable in EDITO pending the new image, but that's still yuck).

This merge request contains the changes that Luiggi did, which are the `AUTOSUBMIT_VERSION` build argument, the conditional to handle that version or use a Git commit or install the latest version, and the change from `dockerfiles` to `docker`, as that's the name of the directory used in AS API & AS GUI (so that it's uniform across projects).

Cheers

**Comment by kinow** [comment]
marked this issue as related to #1372

**Comment by kinow** [comment]
mentioned in commit 2fc2a1bdae477114d7745c474231a326a227106b

**Comment by kinow** [comment]
mentioned in commit a6c337401637bbe2342e09eb1b3d5465ebd1156b

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 31, 2024, 11:20

> @edgano, for the previous image, 4.1.7, I used Luiggi's code, applying a patch manually and then publishing the image (we were in a hurry due to a deliverable in EDITO pending the new image, but that's still yuck).

Do you refer to the image in the DataLab? 

For me, Autosubmit says it's running Autosubmit 4.0.105 like the Jupyterlab-autosubmit-bsc image.

![Screenshot_from_2024-07-31_11-19-39](https://earth.bsc.es/gitlab/es/autosubmit/uploads/eb9f39579d28a0dae1387eca7369c9b3/Screenshot_from_2024-07-31_11-19-39.png)

![Screenshot_from_2024-07-31_11-19-07](https://earth.bsc.es/gitlab/es/autosubmit/uploads/b8c2c08c0e846578cbb88afd1dfb6a52/Screenshot_from_2024-07-31_11-19-07.png)

![Screenshot_from_2024-07-31_11-18-32](https://earth.bsc.es/gitlab/es/autosubmit/uploads/3be1c5966cb3d775bf9a8a92a4977721/Screenshot_from_2024-07-31_11-18-32.png)

**Comment by kinow** [comment]
That was for the datalab, but we didn't use the autosubmit/autosubmit image there, we used autosubmit/edito-demo, which is idneed using 4.0.105.

Will sort out this mess in another issue. For this one I believe we can close as @edgano reviewed it, and we published it to DockerHub. MOi tested and their workflow still failed, but it is not related to the AS version, nor - apparently - to this container.

Thanks Miguel!
